src/main/java/picard/sam/markduplicates/util/AbstractMarkDuplicatesCommandLineProgram.java

```diff
@@ -311,5 +338,27 @@ private static void trackOpticalDuplicates(final List<? extends ReadEnds> list,
         if (opticalDuplicates > 0) {
             opticalDuplicatesByLibraryId.increment(list.get(0).getLibraryId(), opticalDuplicates);
         }
+        return opticalDuplicates;
     }
+
+    private static void trackDuplicateCounts(final int listSize,
+                                             final int optDupCnt,
+                                             final Histogram<Double> duplicatesCountHist,
```

Comment:
It would be less typing to make this accept `LibraryIdGenerator` and have this method call `getDuplicateCountHist()` etc itself.